plugins/out_s3/s3.c

```diff
@@ -119,6 +120,36 @@ static char *mock_error_response(char *error_env_var)
     return NULL;
 }
 
+static void s3_retry_warn(struct flb_s3 *ctx, char* tag, char* input_name, time_t create_time, bool less_than_limit){
+    struct tm now_time;
+    char create_time_str[20];
+    strftime(create_time, 20, "%Y-%m-%d %H:%M:%S", localtime_r(&create_time, &now_time));
+    if (strlen(input_name) != 0) {
```

Comment:
if input_name is not set, will its length be 0? I think it would be NULL? And so then this could be the source of segfault. Its best to check first if its NULL and then check length greater than 0.